### PR TITLE
chore: add generated files for jwt role

### PR DESCRIPTION
**Pull request by RonnanSouza**
### Description of your changes

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/jwt/jwt_test.go [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			e := external{client: tc.fields.clientBuilder(t)}
+			got, err := e.Create(tc.args.ctx, tc.args.mg)
+			if diff := cmp.Diff(tc.want.err, err, test.EquateErrors()); diff != "" {
+				t.Errorf("\n%s\ne.Observe(...): -want error, +got error:\n%s\n", tc.reason, diff)
```
Comment (beuleal) [inline internal/controller/jwt/jwt_test.go]: should it be Create instead Observe? The next line too

File: internal/controller/jwt/jwt_test.go [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			e := external{client: tc.fields.clientBuilder(t)}
+			got, err := e.Update(tc.args.ctx, tc.args.mg)
+			if diff := cmp.Diff(tc.want.err, err, test.EquateErrors()); diff != "" {
+				t.Errorf("\n%s\ne.Observe(...): -want error, +got error:\n%s\n", tc.reason, diff)
```
Comment (beuleal) [inline internal/controller/jwt/jwt_test.go]: Same typo issue s/Observe/Update

File: internal/controller/jwt/jwt_test.go [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			e := external{client: tc.fields.clientBuilder(t)}
+			err := e.Delete(tc.args.ctx, tc.args.mg)
+			if diff := cmp.Diff(tc.want.err, err, test.EquateErrors()); diff != "" {
+				t.Errorf("\n%s\ne.Observe(...): -want error, +got error:\n%s\n", tc.reason, diff)
```
Comment (beuleal) [inline internal/controller/jwt/jwt_test.go]: s/Observe/Delete

**Comment by beuleal** [comment]
Don't we need to validate fields between fields?

**Comment by RonnanSouza** [comment]
> Don't we need to validate fields between fields?

Actually, I'm doing the validations in the CRD itself.

File: internal/controller/jwt/vault.go [hunk 4]
```diff
@@ -40,6 +40,25 @@ type Role struct {
 	TokenType            string                 `json:"token_type"`
 }
 
+// Validate validates if a role follow vault constraints
+// Check https://developer.hashicorp.com/vault/api-docs/auth/jwt#create-role to see vault contraints for JWT/OIDC roles
+func (role *Role) Validate() error {
+	if role.RoleType == "jwt" && len(role.BoundAudiences) == 0 {
```
Comment (beuleal) [inline internal/controller/jwt/vault.go]: As the first part of each `if` statement are the same for all validations, we can refactor them, example:
```
if role.RoleType == "jwt" {
    if len(...) == 0 {
      return error
   }
   if...
   if...

}
```

**Review by beuleal — CHANGES_REQUESTED** [review]

**Comment by beuleal** [comment]
LGTM! Great work buddy!

**Review by beuleal — APPROVED** [review]

**Pull request merged**